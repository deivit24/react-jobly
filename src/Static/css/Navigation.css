nav {
  box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14),
    0 3px 1px -2px rgba(0, 0, 0, 0.12), 0 1px 5px 0 rgba(0, 0, 0, 0.2);
}

.nav-item {
  width: 100px;
  text-align: center;
}

nav a {
  color: #282c34;
  text-decoration: none;
}

nav a.active {
  color: #000;
  font-weight: bold;

  box-sizing: border-box;
}

nav a.active:hover {
  color: #000 !important;
}

nav a:hover {
  text-decoration: none;
  color: #000;
}

nav a.active.navbar-brand {
  border-bottom: none;
  color: #000;
  font-weight: bold;
}

nav a:not(.navbar-brand):hover {
  font-weight: bold;
}

nav a:not(.logout):hover {
  font-weight: bold;
}

nav a.active.logout {
  border-bottom: none;
  color: #282c34;
  font-weight: 400;
}
