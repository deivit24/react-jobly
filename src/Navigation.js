import React, { useState, useContext } from 'react';
import './Static/css/Navigation.css';
import { NavLink } from 'react-router-dom';
import { Collapse, Navbar, NavbarToggler, Nav, NavItem } from 'reactstrap';
import UserContext from './Helpers/UserContext';

const Navigation = ({ logout }) => {
  const { currentUser } = useContext(UserContext);

  const [isOpen, setIsOpen] = useState(false);

  const toggle = () => setIsOpen(!isOpen);

  const loggedIn = (
    <>
      <NavItem>
        <NavLink to="/companies">Companies</NavLink>
      </NavItem>
      <NavItem>
        <NavLink to="/jobs">Jobs</NavLink>
      </NavItem>
      <NavItem>
        <NavLink to="/profile">Profile</NavLink>
      </NavItem>
      <NavItem>
        <NavLink className="logout" to="/" onClick={logout}>
          Logout
        </NavLink>
      </NavItem>
    </>
  );

  const loggedOut = (
    <>
      <NavItem>
        <NavLink to="/login">Login</NavLink>
      </NavItem>
    </>
  );
  return (
    <div className="sticky-top">
      <Navbar color="white" light expand="md">
        <NavLink exact to="/" className="navbar-brand">
          Jobly
        </NavLink>
        <NavbarToggler onClick={toggle} />
        <Collapse isOpen={isOpen} navbar>
          <Nav className="ml-auto" navbar>
            {currentUser ? loggedIn : loggedOut}
          </Nav>
        </Collapse>
      </Navbar>
    </div>
  );
};

export default Navigation;
