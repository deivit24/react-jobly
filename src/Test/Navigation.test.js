import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import Navigation from '../Navigation';
import { UserProvider } from '../testUtils';

describe('Testing the Navigation', () => {
  it('renders without crashing', () => {
    render(
      <MemoryRouter>
        <UserProvider>
          <Navigation />
        </UserProvider>
      </MemoryRouter>
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <UserProvider>
          <Navigation />
        </UserProvider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot when logged out', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <UserProvider currentUser={null}>
          <Navigation />
        </UserProvider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
