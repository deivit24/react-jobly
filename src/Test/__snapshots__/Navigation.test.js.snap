// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Navigation matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sticky-top"
  >
    <nav
      class="navbar navbar-expand-md navbar-light bg-white"
    >
      <a
        aria-current="page"
        class="navbar-brand active"
        href="/"
      >
        Jobly
      </a>
      <button
        aria-label="Toggle navigation"
        class="navbar-toggler"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
      >
        <ul
          class="ml-auto navbar-nav"
        >
          <li
            class="nav-item"
          >
            <a
              href="/companies"
            >
              Companies
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              href="/jobs"
            >
              Jobs
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              href="/profile"
            >
              Profile
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              aria-current="page"
              class="logout active"
              href="/"
            >
              Logout
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`Testing the Navigation matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div
    class="sticky-top"
  >
    <nav
      class="navbar navbar-expand-md navbar-light bg-white"
    >
      <a
        aria-current="page"
        class="navbar-brand active"
        href="/"
      >
        Jobly
      </a>
      <button
        aria-label="Toggle navigation"
        class="navbar-toggler"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
      >
        <ul
          class="ml-auto navbar-nav"
        >
          <li
            class="nav-item"
          >
            <a
              href="/login"
            >
              Login
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`Testing the Routes matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sticky-top"
  >
    <nav
      class="navbar navbar-expand-md navbar-light bg-white"
    >
      <a
        aria-current="page"
        class="navbar-brand active"
        href="/"
      >
        Jobly
      </a>
      <button
        aria-label="Toggle navigation"
        class="navbar-toggler"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
      >
        <ul
          class="ml-auto navbar-nav"
        >
          <li
            class="nav-item"
          >
            <a
              href="/companies"
            >
              Companies
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              href="/jobs"
            >
              Jobs
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              href="/profile"
            >
              Profile
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              aria-current="page"
              class="logout active"
              href="/"
            >
              Logout
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`Testing the Routes matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div
    class="sticky-top"
  >
    <nav
      class="navbar navbar-expand-md navbar-light bg-white"
    >
      <a
        aria-current="page"
        class="navbar-brand active"
        href="/"
      >
        Jobly
      </a>
      <button
        aria-label="Toggle navigation"
        class="navbar-toggler"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
      >
        <ul
          class="ml-auto navbar-nav"
        >
          <li
            class="nav-item"
          >
            <a
              href="/login"
            >
              Login
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
